src/paging.rs

```diff
@@ -60,6 +62,36 @@ impl TranslationRegime {
     pub(crate) fn supports_asid(self) -> bool {
         matches!(self, Self::El2And0 | Self::El1And0)
     }
+
+    /// Invalidate the memory range starting at `va`. The size of the range is unspecified, it is
+    /// up to the caller to apply this function as needed to invalidate a range of memory
+    pub(crate) fn invalidate_va(self, va: VirtualAddress) {
+        #[allow(unused)]
+        let va = va.0 >> 12;
+
+        // SAFETY: TLBI maintenance has no side effects that are observeable by the
+        // program
+        #[cfg(all(not(test), target_arch = "aarch64"))]
+        unsafe {
+            match self {
+                TranslationRegime::El1And0 => asm!(
+                    "tlbi vaae1is, {va}",
+                    va = in(reg) va,
+                    options(preserves_flags),
+                ),
+                TranslationRegime::El3 => asm!(
+                    "tlbi vae3is, {va}",
+                    va = in(reg) va,
+                    options(preserves_flags),
+                ),
+                _ => asm!(
```

Comment:
Let's list the remaining two translation regimes explicitly here for this match arm, so we don't forget to add an arm if more translation regimes are added in future.